src/lua/uri.lua

```diff
@@ -292,31 +292,34 @@ local function unescape(buf, opts)
     return ffi.string(dst, dst_size)
 end
 
-local function encode_kv(key, values, res)
+local function encode_kv(key, values, res, escape_opts)
     local val = values
     if type(val) ~= "table" then
         val = { val }
     end
+    local key_escaped= escape(key, escape_opts)
```

Comment:
That's interesting: you call `tostring()` on values, but not on keys. I think that we should follow the same rule here. I guess that the idea behind this was to accept numbers in values. Moreover, it is more important for keys, because `{'a', 'b', 'c'}` has numeric keys.

- [ ] Let's accept them in values and keys both (call `tostring()` for both).
- [ ] Let's add test cases with numeric keys and values.

Reply:
Ouch, I didn't know that everything except strings is forbidden for keys. However allowing at least numbers seems correct for me.